### PR TITLE
Fix maven pom resource assignment

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
This introduces a fix where we don't assign resources to packages when we
find multiple pom.xmls in a META-INF directory for a jar.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [x] Looked for possible updates in documentation and added updates if applicable
* [ ] Updated CHANGELOG.rst

### CODE REVIEW

**Comment by AyanSinhaMahapatra** [comment]
@pombredanne ready for initial review, please take a look

File: src/packagedcode/maven.py [hunk 1]
```diff
@@ -132,10 +132,21 @@ def assign_package_to_resources(cls, package, resource, codebase, package_adder)
...
             # /META-INF/maven/log4j/log4j/pom.xml: This requires going up 5 times
-            upward_segments = 5
+            upward_segments = 4
             root = resource
             for _ in range(upward_segments):
                 root = root.parent(codebase)
+
+            number_poms = 0
+            for child in root.walk(codebase):
+                if 'pom.xml' in child.path:
```
Comment (pombredanne) [inline src/packagedcode/maven.py:142]:
```suggestion
                if child.path.endswith('pom.xml'):
```

**Review by pombredanne — APPROVED** [review]
LGTM with a minor nit.... You may want to add a test for the case when there is single pom too

File: src/packagedcode/maven.py (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply src/packagedcode/maven.py:142]: @pombredanne there were other cases already for this, so this is tested.

**Pull request merged**